### PR TITLE
Add FV tests for WireGuard host to workload

### PR DESCRIPTION
**Pull request by neiljerram**
## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: fv/wireguard_test.go [hunk 1]
```diff
@@ -820,6 +824,37 @@ var _ = infrastructure.DatastoreDescribe("WireGuard-Supported 3 node cluster", [
 				}, "10s", "100ms").Should(BeNumerically(">", 0))
 			}
 		})
+
+		It("WireGuard should be used for host to workload connections on WireGuard enabled nodes", func() {
+			cc.ExpectSome(felixes[0], wls[1])
```
Comment (realgaurav) [inline fv/wireguard_test.go:829]: `cc.ExpectSome(felixes[1], wls[0])`?

File: fv/wireguard_test.go [hunk 2]
```diff
@@ -820,6 +824,37 @@ var _ = infrastructure.DatastoreDescribe("WireGuard-Supported 3 node cluster", [
...
+			}
+		})
+
+		It("WireGuard should not be used for host to workload connections when WireGuard disabled on either node", func() {
+			cc.ExpectSome(felixes[0], wls[2])
+			cc.ExpectSome(felixes[2], wls[0])
+			cc.CheckConnectivity()
+
+			By("verifying packets between felix-0 and felix-2 are not encrypted")
+			for i := range []int{0, 2} {
```
Comment (realgaurav) [inline fv/wireguard_test.go:849]: Should this be?

`for _, i := range []int{0, 2} {`

File: fv/wireguard_test.go (hunk 1 above)
Comment (neiljerram) [reply fv/wireguard_test.go:829]: No; as that should be exactly symmetric, I thought the encrypted packets testing would be more precise if it was done with just one attempted connection.

File: fv/wireguard_test.go (hunk 2 above)
Comment (neiljerram) [reply fv/wireguard_test.go:849]: Thanks, good catch!

**Pull request merged**